src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1656,26 +1643,46 @@
                    unexpected))))
 
     (if-let [pending (seq (pending-migrations))]
-      (let [tables-to-analyze (jdbc/with-db-transaction []
-                                (->> pending
-                                     (map (fn [[version migration]]
-                                            (let [t0 (now)]
-                                            (log/info (trs "Applying database migration version {0}" version))
-                                            (sql-or-die (fn [] (let [result (migration)]
-                                                                    (record-migration! version)
-                                                                    result)))
-                                            (log/info (trs "Applied database migration version {0} in {1} ms"
-                                                           version (in-millis (interval t0 (now))))))))
-                                     (filter map?)
-                                     (map ::vacuum-analyze)
-                                     (apply set/union small-tables)
-                                     sort))]
-        (log/info (trs "Updating table statistics for: {0}" (str/join ", " tables-to-analyze)))
+      (let [tables (jdbc/with-db-transaction []
+                     (->> pending
+                          (map (fn [[version migration]]
+                                 (log/info (trs "Applying database migration version {0}" version))
+                                 (let [t0 (now)]
+                                   (let [result (migration)]
+                                     (record-migration! version)
+                                     (log/info (trs "Applied database migration version {0} in {1} ms"
+                                                    version (in-millis (interval t0 (now)))))
+                                     result))))
+                          (filter map?)
+                          (map ::vacuum-analyze)
+                          (apply set/union small-tables)))]
+        (analyze tables)
         true)
       (do
         (log/info (trs "There are no pending migrations"))
+        ;; Always analyze these since we had a good long period where
+        ;; the post-migration analysis above wasn't actually working,
+        ;; and since this shouldn't be expensive.
+        (analyze small-tables)
         false))))
 
+(defn migrate! [db-connection-pool]
+  "Migrates database to the latest schema version. Does nothing if
+  database is already at the latest schema version.  Requires a
+  connection pool because some operations may require an indepdendent
+  database connection.
+
+  This function will return true iff any migrations were run."
```

Comment:
s/iff/if/

Reply:
Hah, oh, that was the "if and only if" (math) abbreviation, which I tend to use, but perhaps (and possibly to your point) I shouldn't here.